Frontend Boilerplate Code

Nailsonseat opened the pull request:
## Front-end boiler plate code (FOSS 2024)

- Created directories like assets, screens, routes etc.
- Added basic dependencies like provider, go router, secure storage.
- package name - com.openlake.smartinstiapp


<br><br>
<table>
  <tr>
    <td>
      <img src="https://github.com/OpenLake/Smart-Insti-App/assets/67797628/0e3495bb-dea0-4e60-a16d-6df719f51c84" alt="Image" width="400">
    </td>
    <td>
      <img src="https://github.com/OpenLake/Smart-Insti-App/assets/67797628/3df64947-baee-4b90-bb8e-c3ddcc35cf65" alt="Image" width="400">
    </td>
  </tr>

henilp105 reviewed (approved):
Thanks @Nailsonseat , Looks good to me , I would like to suggest some stylistic changes:
1. the directories should be named as the frontend and backend directories.
2. there should be consolidated `gitignore` for the entire frontend instead of android, ios. 
3. It would be great if there is a github action to build the app and store it in archives.

Let's wait for an annother review then I will merge this. 

CC: @tusharbansal22

Nailsonseat commented:
@henilp105 we have added some of your suggestions in this PR itself

- [x] renamed to frontend directory. Backend will be renamed too in #10  
- [x] added a global git ignore in root which overlooks both front and backend, making it easier to read and write.
- [ ]  WIll add a github action in future PR to build an apk, for easier testing and reviewing

All the above changes have been tested by our team members and produce the same screenshots as above.

henilp105 reviewed (approved):
Thanks @Nailsonseat , looks good to me merging! Congratulations for your first PR for FossOverflow.
